examples/api/lib/cupertino/modal_bottom_sheet/modal_bottom_sheet.0.dart

```diff
@@ -0,0 +1,373 @@
+      parentNavigatorContext: context,
+      initialRoute: '/',
+      onGenerateRoute: (RouteSettings settings) {
+        return CupertinoPageRoute(
+          builder: (BuildContext context) {
+            return Semantics(
+              scopesRoute: true,
+              explicitChildNodes: true,
+              child: child,
+            );
+          }
+        );
+      }
+    );
+  }
+
+  /// Returns a [CupertinoFullscreenDialogTransition] if [route] is a full
+  /// screen dialog, otherwise a [CupertinoPageTransition] is returned.
+  ///
+  /// Used by [CupertinoPageRoute.buildTransitions].
+  ///
+  /// This method can be applied to any [PageRoute], not just
+  /// [CupertinoPageRoute]. It's typically used to provide a Cupertino style
+  /// horizontal transition for material widgets when the target platform
+  /// is [TargetPlatform.iOS].
+  ///
+  /// See also:
+  ///
+  ///  * [CupertinoPageTransitionsBuilder], which uses this method to define a
+  ///    [PageTransitionsBuilder] for the [PageTransitionsTheme].
+  static Widget buildPageTransitions<T>(
+    ModalRoute<T> route,
+    BuildContext context,
+    Animation<double> animation,
+    Animation<double> secondaryAnimation,
+    Widget child,
+  ) {
+    return MBSTransition(
+      primaryRouteAnimation: animation,
+      secondaryRouteAnimation: secondaryAnimation,
+      child: child,
+    );
+  }
+
+  @override
+  Widget buildTransitions(BuildContext context, Animation<double> animation, Animation<double> secondaryAnimation, Widget child) {
+    return buildPageTransitions<T>(this, context, animation, secondaryAnimation, child);
+  }
+}
+
+class MBSNavigator extends Navigator {
+
+  MBSNavigator({
+    super.key,
+    super.pages,
+    super.onPopPage,
+    super.initialRoute,
+    super.onGenerateInitialRoutes,
+    super.onGenerateRoute,
+    super.onUnknownRoute,
+    super.transitionDelegate,
+    super.reportsRouteUpdateToEngine,
+    super.clipBehavior,
+    super.observers,
+    super.requestFocus,
+    super.restorationScopeId,
+    super.routeTraversalEdgeBehavior,
+    required this.parentNavigatorContext,
+  });
+
+  final BuildContext parentNavigatorContext;
+
+  void popMBS() {
+    Navigator.of(parentNavigatorContext).pop();
+    // Navigator.of(parentNavigatorContext).delegateTransitionBuilder = null;
+  }
+
+  static NavigatorState? of(
+    BuildContext context, {
+    bool rootNavigator = false,
+  }) {
+    // Handles the case where the input context is a navigator element.
+    NavigatorState? navigator;
+    if (context is StatefulElement && context.state is NavigatorState) {
+      navigator = context.state as NavigatorState;
+    }
+    if (rootNavigator) {
+      navigator = context.findRootAncestorStateOfType<NavigatorState>() ?? navigator;
+    } else {
+      navigator = navigator ?? context.findAncestorStateOfType<NavigatorState>();
+    }
+    if (navigator?.widget is! MBSNavigator) {
+      return null;
+    }
+
+    return navigator;
+  }
+}
```

Comment:
A lot of the code in this file would actually end up in a package, so what would the end-user experience be for using MBS in this case? And how does it compare to the current user experience for modal_bottom_sheet etc.? It looks like users would have to use a custom page route, but I guess that's similar to modal_bottom_sheet's MaterialWithModalsPageRoute.